crypto/src/main/kotlin/web5/sdk/crypto/AwsKeyManager.kt

```diff
@@ -0,0 +1,200 @@
+package web5.sdk.crypto
+
+import com.amazonaws.services.kms.AWSKMS
+import com.amazonaws.services.kms.AWSKMSClientBuilder
+import com.amazonaws.services.kms.model.AWSKMSException
+import com.amazonaws.services.kms.model.CreateAliasRequest
+import com.amazonaws.services.kms.model.CreateKeyRequest
+import com.amazonaws.services.kms.model.DescribeKeyRequest
+import com.amazonaws.services.kms.model.GetPublicKeyRequest
+import com.amazonaws.services.kms.model.KeySpec
+import com.amazonaws.services.kms.model.KeyUsageType
+import com.amazonaws.services.kms.model.MessageType
+import com.amazonaws.services.kms.model.SignRequest
+import com.amazonaws.services.kms.model.SigningAlgorithmSpec
+import com.nimbusds.jose.Algorithm
+import com.nimbusds.jose.JWSAlgorithm
+import com.nimbusds.jose.crypto.impl.ECDSA
+import com.nimbusds.jose.jwk.Curve
+import com.nimbusds.jose.jwk.ECKey
+import com.nimbusds.jose.jwk.JWK
+import com.nimbusds.jose.jwk.KeyUse
+import org.bouncycastle.asn1.x509.SubjectPublicKeyInfo
+import org.bouncycastle.crypto.ExtendedDigest
+import org.bouncycastle.crypto.digests.SHA256Digest
+import org.bouncycastle.crypto.digests.SHA384Digest
+import org.bouncycastle.crypto.digests.SHA512Digest
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter
+import java.nio.ByteBuffer
+import java.security.PublicKey
+import java.security.interfaces.ECPublicKey
+
+/**
+ * A [KeyManager] that uses AWS KMS for remote storage of keys and signing operations. Caller is expected to provide
+ * connection details for [AWSKMS] client as per [Configure the AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html)
+ *
+ * Key aliases are generated from the key's JWK thumbprint, and stored in AWS KMS.
+ * e.g. alias/6uNnyj7xZUgtKTEOFV2mz0f7Hd3cxIH1o5VXsOo4u1M
+ *
+ * AWS supports a limited set ECDSA curves for signing:
+ * - [JWSAlgorithm.ES256K]
+ * - [JWSAlgorithm.ES256]
+ * - [JWSAlgorithm.ES384]
+ * - [JWSAlgorithm.ES512]
+ */
+public class AwsKeyManager : KeyManager {
+
+  private val kmsClient: AWSKMS = AWSKMSClientBuilder.standard().build()
```

Comment:
note from @bradleydwyer: might be worth including ability to optionally provide AWS KMS Client configuration